Src/library/memory_analyzer/singletons/ma_shared_main_window.e

```diff
@@ -0,0 +1,68 @@
+note
+	description: "Summary description for {MA_SHARED_MAIN_WINDOW}."
+	author: ""
+	date: "$Date$"
+	revision: "$Revision$"
+
+class
+	MA_SHARED_MAIN_WINDOW
+
+feature -- Access
+
+	main_window: detachable MA_WINDOW
```

Comment:
Note this new class is just the previous design (in MA_SINGLETON_FACTORY) splitted into two classes.
I added this class, since it was not really the same nature of other once from MA_SINGLETON_FACTORY, and I guess the MA_SINGLETON_FACTORY.main_window was more a workaround due to void-safety and the difficulty to use Current to initialize attached attributes.